File: guanabara-java/src/com/danillo/algoritmos/lista04/Exercicio55.java
package com.danillo.algoritmos.lista04;

import java.util.Random;
import java.util.Scanner;

public class Exercicio55 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        Random aleatorio = new Random();

        int i=0, numero = aleatorio.nextInt(10);

        while(i < 4){
            System.out.println("Tente adivinhar o número mágico de 1 a 10: ");
            int chute = entrada.nextInt();

            if (chute == numero){
                System.out.println("Acertou!");
                break;
            } else {
                System.out.println("Errou!");
            }

            i++;
        }
    }
}

File: guanabara-python/src/lista04/exercicio55.py
import random

numero_magico = random.randint(1, 10)

i = 0
while i <= 4:
    x = int(input("Tente acertar o número mágico: "))

    if x == numero_magico:
        print(f"Acertou! O número mágico é {numero_magico}")
        break
    else:
        print("Errou!")

    i = i + 1
